model/channel_search.go

```diff
@@ -11,11 +11,18 @@ import (
 const CHANNEL_SEARCH_DEFAULT_LIMIT = 50
 
 type ChannelSearch struct {
-	Term                   string `json:"term"`
-	ExcludeDefaultChannels bool   `json:"exclude_default_channels"`
-	NotAssociatedToGroup   string `json:"not_associated_to_group"`
-	Page                   *int   `json:"page,omitempty"`
-	PerPage                *int   `json:"per_page,omitempty"`
+	Term                    string   `json:"term"`
+	ExcludeDefaultChannels  bool     `json:"exclude_default_channels"`
+	NotAssociatedToGroup    string   `json:"not_associated_to_group"`
+	TeamIds                 []string `json:"team_ids"`
+	GroupConstrained        bool     `json:"group_constrained"`
+	ExcludeGroupConstrained bool     `json:"exclude_group_constrained"`
+	Public                  bool     `json:"public"`
+	Private                 bool     `json:"private"`
+	IncludeDeleted          bool     `json:"include_deleted"`
+	Deleted                 bool     `json:"deleted"`
```

Comment:
I have some concerns about these being value fields. This means that they will fall back to their default values if in any case, the UI forgets to set them in the HTTP request. I see that in the code you are only checking for non-default values, but I am a bit concerned if in the future, somebody adds a condition which checks for the negative value for these fields.

Are all code paths from the app explicitly set these fields to either true or false?

Reply:
The default code path sets false for all of the above options.